test/test_utils.py

```diff
@@ -217,6 +217,36 @@ def objective_h(x):
         return x
 
 
+def test_cache_decorator_does_not_compare_infinite_return_values():
+    cache_fn = tempfile.mkstemp()[1]
+
+    @cgp.utils.disk_cache(cache_fn)
+    def objective_f(x):
+        try:
+            return 1.0 / x
+        except ZeroDivisionError:
+            return np.inf
+
+    # first call produces infinite return value, identical to
+    # objective_g although in general their return values are
+    # different
+    objective_f(0.0)
+    # second call produces a finite return value which should be used
+    # to check consistency
+    objective_f(2.0)
+
+    # since the consistency check uses the finite return value it
+    # should detect that the two objectives are indeed different
```

Comment:
```suggestion
```

Reply:
I think this is clear without the comment

Reply:
i agree, but in tests i tend to favor overdocumentation. not sure whether this is best practice though.